packages/swap/src/routes/v2/quote.ts

```diff
@@ -0,0 +1,370 @@
+
+  return {
+    chunkSize: BigInt(new BigNumber(amount.toString()).dividedBy(numberOfChunks).toFixed(0)),
+    numberOfChunks,
+    addedDurationSeconds: Math.ceil(env.DCA_CHUNK_INTERVAL_BLOCKS * 6 * (numberOfChunks - 1)), // we deduct 1 chunk because the first one is already accounted for in the regular quote
+  };
+};
+
+const adjustDcaQuote = ({
+  dcaQuoteParams,
+  dcaQuote,
+  dcaBoostedQuote,
+  estimatedBoostFeeBps,
+}: {
+  dcaQuoteParams: NonNullable<Awaited<ReturnType<typeof getDcaQuoteParams>>>;
+  dcaQuote: QuoteQueryResponse;
+  dcaBoostedQuote?: QuoteQueryResponse | null | 0;
+  estimatedBoostFeeBps?: number;
+}) => {
+  // eslint-disable-next-line no-param-reassign
+  dcaQuote.dcaParams = {
+    chunkInterval: env.DCA_CHUNK_INTERVAL_BLOCKS,
+    numberOfChunks: dcaQuoteParams.numberOfChunks,
+  };
+  if (dcaQuoteParams && dcaQuote) {
+    const netWorkFee = dcaQuote.includedFees.find((fee) => fee.type === 'NETWORK');
+    if (netWorkFee) {
+      netWorkFee.amount = new BigNumber(netWorkFee.amount)
+        .multipliedBy(dcaQuoteParams.numberOfChunks)
+        .toFixed(0);
+    }
+
+    const brokerFee = dcaQuote.includedFees.find((fee) => fee.type === 'BROKER');
+    if (brokerFee) {
+      brokerFee.amount = new BigNumber(brokerFee.amount)
+        .multipliedBy(dcaQuoteParams.numberOfChunks)
+        .toFixed(0);
+    }
+
+    // eslint-disable-next-line no-param-reassign
+    dcaQuote.egressAmount = new BigNumber(dcaQuote.egressAmount)
+      .multipliedBy(dcaQuoteParams.numberOfChunks)
+      .toFixed(0);
+    // eslint-disable-next-line no-param-reassign
+    dcaQuote.estimatedDurationSeconds += dcaQuoteParams.addedDurationSeconds;
+  }
+  if (dcaQuoteParams && dcaBoostedQuote && estimatedBoostFeeBps) {
+    const netWorkFee = dcaBoostedQuote.includedFees.find((fee) => fee.type === 'NETWORK');
+    if (netWorkFee) {
+      netWorkFee.amount = new BigNumber(netWorkFee.amount)
+        .multipliedBy(dcaQuoteParams.numberOfChunks)
+        .toFixed(0);
+    }
+
+    const brokerFee = dcaBoostedQuote.includedFees.find((fee) => fee.type === 'BROKER');
+    if (brokerFee) {
+      brokerFee.amount = new BigNumber(brokerFee.amount)
+        .multipliedBy(dcaQuoteParams.numberOfChunks)
+        .toFixed(0);
+    }
+
+    const boostFee = dcaBoostedQuote.includedFees.find((fee) => fee.type === 'BOOST');
+    if (boostFee) {
+      boostFee.amount = new BigNumber(boostFee.amount)
+        .multipliedBy(dcaQuoteParams.numberOfChunks)
+        .toFixed(0);
+    }
+
+    // eslint-disable-next-line no-param-reassign
+    dcaQuote.boostQuote = {
+      ...dcaBoostedQuote,
+      estimatedBoostFeeBps,
+      egressAmount: BigNumber(dcaBoostedQuote.egressAmount)
+        .multipliedBy(dcaQuoteParams.numberOfChunks)
+        .toFixed(0),
+      estimatedDurationSeconds:
+        dcaBoostedQuote.estimatedDurationSeconds + dcaQuoteParams.addedDurationSeconds,
+    };
+  }
+};
+
+export const validateQuoteQuery = async (query: Query) => {
```

Comment:
i would move these shared functions into its own file

Reply:
as discussed offline, will do this in a later PR